packages/tx/src/eip4844Transaction.ts

```diff
@@ -0,0 +1,479 @@
+import { byteArrayEquals, toHexString } from '@chainsafe/ssz'
+import {
+  Address,
+  MAX_INTEGER,
+  bigIntToHex,
+  bigIntToUnpaddedBuffer,
+  bufferToBigInt,
+  bufferToHex,
+  computeVersionedHash,
+  ecrecover,
+  toBuffer,
+} from '@ethereumjs/util'
+import { keccak256 } from 'ethereum-cryptography/keccak'
+
+import { BaseTransaction } from './baseTransaction'
+import { kzg } from './kzg/kzg'
+import {
+  BLOB_COMMITMENT_VERSION_KZG,
+  BlobNetworkTransactionWrapper,
+  BlobTransactionType,
+  LIMIT_BLOBS_PER_TX,
+  SignedBlobTransactionType,
+} from './types'
+import { AccessLists, checkMaxInitCodeSize } from './util'
+
+import type {
+  AccessList,
+  AccessListBuffer,
+  AccessListBufferItem,
+  BlobEIP4844TxData,
+  JsonTx,
+  TxOptions,
+  TxValuesArray,
+} from './types'
+import type { Common } from '@ethereumjs/common'
+
+const TRANSACTION_TYPE = 0x05
+const TRANSACTION_TYPE_BUFFER = Buffer.from(TRANSACTION_TYPE.toString(16).padStart(2, '0'), 'hex')
+
+const validateBlobTransactionNetworkWrapper = (
+  versionedHashes: Uint8Array[],
+  blobs: Uint8Array[],
+  commitments: Uint8Array[],
+  kzgProof: Uint8Array
+) => {
+  if (!(versionedHashes.length === blobs.length && blobs.length === commitments.length)) {
+    throw new Error('Number of versionedHashes, blobs, and commitments not all equal')
```

Comment:
Not sure if practically relevant: maybe do this in two separate checks with dedicated error messages and not mix here (unnecessarily (?))? In doubt if being a "can't happen" (just one of these errors at a time) case just ignore.

Reply:
I think this check should all be one because if the count for any of the three doesn't match, it's an invalid transaction.

Reply:
Ok 👍